core/store/src/trie/mem/arena/alloc.rs

```diff
@@ -56,8 +58,8 @@ const NUM_ALLOCATION_CLASSES: usize = allocation_class(MAX_ALLOC_SIZE) + 1;
 impl Allocator {
     pub fn new(name: String) -> Self {
         Self {
-            freelists: [usize::MAX; NUM_ALLOCATION_CLASSES],
-            next_ptr: 0,
+            freelists: [ArenaPos::invalid(); NUM_ALLOCATION_CLASSES],
+            next_ptr: ArenaPos::invalid(),
```

Comment:
I think allocating the first chunk here and setting next_ptr to its beginning would slightly simplify the code, would it not? (by avoiding `is_invalid` checks)